WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -93,6 +92,22 @@ protected void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    private void trackResultAndFinish() {
+        if (!TextUtils.isEmpty(reason)) {
+            AppLog.e(AppLog.T.API, "Could not connect to Jetpack, reason: "+reason);
+            ToastUtils.showToast(this, reason);
+        } else {
+            trackWithSource(Stat.SIGNED_INTO_JETPACK, source);
+        }
+        if (source == Source.STATS) {
+            //Reload the site to get new stats
+            reloadSite();
+        } else {
```

Comment:
I am not sure if I understand why we need to wait for the reload of the site to be done. I feel like we should be reloading the site unless there is an error, even if the source was not the stats, because we know the site has changed. However, once we dispatch this, we don't actually need to wait for the callback. If the stats page needs to be reloaded when the site changed, it should handle it, this activity shouldn't need to do anything.

Reply:
In theory you're right. The Stats page doesn't handle this well though. If the sync is not fast enough, it will still think that the Jetpack is not connected and open the `Connect Jetpack` activity. Do you see a clean way out of this? Stats activity is pretty messy.